fiona/__init__.py

```diff
@@ -67,21 +67,67 @@
 import os
 import sys
 import warnings
-
+import platform
 from six import string_types
 from collections import OrderedDict
-
-try:
-    from pathlib import Path
-except ImportError:  # pragma: no cover
-    class Path:
-        pass
+from pathlib import Path
 
 # TODO: remove this? Or at least move it, flake8 complains.
 if sys.platform == "win32":
     libdir = os.path.join(os.path.dirname(__file__), ".libs")
     os.environ["PATH"] = os.environ["PATH"] + ";" + libdir
 
+
+try:
+
+    import fiona.ogrext
+
+except ImportError as e:
+    """
+    With Python >= 3.8 on Windows directories in PATH are not automatically
+    searched for DLL dependencies and must be added manually with
+    os.add_dll_directory.
+
+    see https://github.com/Toblerity/Fiona/issues/851
+    """
+    if platform.system() == 'Windows':
+        
+        def add_dll_directory_win():
+            """
+                Check if a */gdal/bin directory is found in PATH.
+                If none is found, use GDAL_HOME if present.
+            """
+
+            dll_directory = None
+
+            # Parse PATH for gdal/bin
+            for _path in os.getenv('PATH', '').split(';'):
+                p = Path(_path.lower())
+
+                if p.parts[-2:] == ('gdal', 'bin') and os.path.exists(_path):
+                    dll_directory = _path
+                    break
+
+            # Use GDAL_HOME if present
+            if dll_directory is not None:
```

Comment:
Does this mean that `GDAL_HOME` is only checked if a directory has already been found in the `PATH`? That doesn't seem right.

Reply:
fixed